### PR TITLE
Generalize MBC constants

### PR DESCRIPTION
**Pull request by tobiasvl**
- Use hardware.inc constants where possible
- Rename MBC3SelectBank to rSelectROMBank (LADX doesn't use MBC3)

### CODE REVIEW

File: src/code/entities/07_horse_piece.asm [hunk 1]
```diff
@@ -72,7 +72,7 @@ HorsePieceEntityHandler::
 ._04 dw HorsePieceStateFinished                   ; $763E
 
 HorsePieceStateInit::
-    jp   HorsePieceCheckForPickup                  ; $7640: $C3 $33 $77
+    jp   HorsePieceCheckForPickup                 ; $7640: $C3 $33 $77
```
Comment (tobiasvl) [inline src/code/entities/07_horse_piece.asm:75]: Rogue misaligned comment caught by `align_comments.py`

**Review by kemenaran — APPROVED** [review]

File: src/code/bank1.asm [hunk 2]
```diff
@@ -1807,7 +1807,7 @@ UpdateRecentRoomsList::
 
 HideAllSprites::
     ; $0000 controls whether to enable external RAM writing
-    ld   hl, MBC3SRamEnable                       ; $5F2E: $21 $00 $00
+    ld   hl, rRAMG                                ; $5F2E: $21 $00 $00
```
[the remark this answers is not in the document]
Comment (kemenaran) [reply src/code/bank1.asm:1810]: Hmm, that's strange…

[the remark this answers is not in the document]
Comment (kemenaran) [reply src/code/bank1.asm:1810]: Maybe it used to copy the content of OAM memory elsewhere, but the current version of the code doesn't. But then why add a specific GBC condition…

Or it is used as a side-effect by the caller?

[the remark this answers is not in the document]
Comment (tobiasvl) [reply src/code/bank1.asm:1810]: Maybe. Just something I noticed though, I'll just merge this now and keep exploring this.

**Pull request merged**